AFMReader/stp.py

```diff
@@ -69,14 +70,20 @@ def load_stp(  # noqa: C901 (ignore too complex)
             if cols_match is None:
                 raise ValueError(f"[{filename}] : 'cols' not found in file header.")
             cols = int(cols_match.group(1))
-            x_real_size_match = re.search(r"X Amplitude: (\d+\.?\d*) nm", header_decoded)
+            x_real_size_match = re.search(r"X Amplitude: (\d+\.?\d*) (µm|nm)", header_decoded)
             if x_real_size_match is None:
                 raise ValueError(f"[{filename}] : 'X Amplitude' not found in file header.")
             x_real_size = float(x_real_size_match.group(1))
-            y_real_size_match = re.search(r"Y Amplitude: (\d+\.?\d*) nm", header_decoded)
+            x_units = x_real_size_match.group(2)
```

Comment:
A matter of style but I find the one line `if ... else...` would be easy to read here (and other places).

```suggestion
            x_units = x_real_size*1000 if x_real_size_match.group(2) == "µm" else x_real_size
```

Don't feel you have to go with this though.

Reply:
Thanks, I'll commit this in both

I haven't - how big is TopoStats?